js/battle.js

```diff
@@ -6104,16 +6104,16 @@ var Battle = (function () {
 				}
 				if (isSwitch && pokemon == this.p1.lastPokemon && !this.p1.active[slot]) continue;
 				if ((searchid && pokemon.searchid === searchid) || // exact match
-					(!pokemon.searchid && pokemon.checkDetails(details)) || // switch-in matches Team Preview entry
 					(!searchid && pokemon.ident === pokemonid)) { // name matched, good enough
-					if (!pokemon.searchid && createIfNotFound) {
-						pokemon.name = name;
-						pokemon.searchid = searchid;
-						pokemon.ident = pokemonid;
-						if (pokemon.needsReplace) {
-							pokemon = this.p1.newPokemon(this.parseDetails(name, pokemonid, details), i);
-						}
-					}
+					if (slot >= 0) pokemon.slot = slot;
+					return pokemon;
+				}
+				if (!pokemon.searchid && (pokemon.checkDetails(details) || // switch-in matches Team Preview entry
+					pokemon.checkDetails(details.replace(', shiny', '')))) { // switch-in matches non-shiny Team Preview entry
+					pokemon.name = name;
+					pokemon.searchid = searchid;
+					pokemon.ident = pokemonid;
+					pokemon = this.p1.newPokemon(this.parseDetails(name, pokemonid, details), i);
```

Comment:
`.needsReplace` isn't always true here - and if it were, the three lines above would be useless...

Reply:
In other words, you need to add back `if (pokemon.needsReplace) {` here.

Reply:
Do I also not have to mark every single Pokemon with `needsReplace` then, since we don't know from Team Preview which of them are shiny or not?

Reply:
Or something like this?

    if (pokemon.needsReplace) {
        ...
    else {
        ...
    }

Reply:
Good point, we can just always replace the Pokemon object. In which case just remove the above three `pokemon.... = ...` lines.

Reply:
And then remove the `needsReplace` flag, since we're just replacing them all anyway.